### PR TITLE
4P5 121/fix motor movement

### PR DESCRIPTION
**Pull request by C0DK**

### CODE REVIEW

File: code/nxt/inc/movement.h [hunk 1]
```diff
@@ -2,22 +2,41 @@
 #define MOVEMENT_H
 
 #include <stdbool.h>
+#include <stdlib.h>
```
Comment (technight) [inline code/nxt/inc/movement.h]: Why include stdlib?

File: code/nxt/inc/movement.h [hunk 2]
```diff
@@ -2,22 +2,41 @@
...
 
 #include "target_location.h"
 
-#define MOTOR_SPEED_UPPER_BOUND 30
-#define MOTOR_SPEED_LOWER_BOUND 15
+#define MOTOR_RANGE_Y 10 // 50
```
Comment (technight) [inline code/nxt/inc/movement.h]: remove comment

Comment (technight) [reply code/nxt/inc/movement.h]: Also below

File: code/nxt/src/movement.c [hunk 3]
```diff
@@ -6,61 +6,27 @@ uint8_t y_motor = 0;
...
-/*            relation to the max_x and max_y values.                       */
-/*--------------------------------------------------------------------------*/
+uint8_t x_lower_bound_modifier = 0;
+uint8_t y_lower_bound_modifier = 0;
 
-bool move_to(T_TARGET_LOCATION target){
-    target_location = target;
-  return true;
-}
+T_TARGET_LOCATION last_location = {0,0,0};
```
Comment (technight) [inline code/nxt/src/movement.c]: Spaces

File: code/nxt/src/movement.c [hunk 4]
```diff
@@ -6,61 +6,27 @@ uint8_t y_motor = 0;
...
-  T_TARGET_LOCATION actual_target_location = {current_location.x + target_location.x,
-                                              current_location.y + target_location.y,
-                                              target_location.timestamp};
-
+void move(T_TARGET_LOCATION target){
   // speed is 0 when distance is small enough.
-  ecrobot_set_motor_speed(x_motor, get_speed_by_distance(actual_target_location.x));
-  ecrobot_set_motor_speed(y_motor, get_speed_by_distance(actual_target_location.y));
-
+  ecrobot_set_motor_speed(x_motor, get_speed_by_distance(target.x,'x'));
```
Comment (technight) [inline code/nxt/src/movement.c]: spaces

File: code/nxt/src/movement.c [hunk 5]
```diff
@@ -128,16 +94,72 @@ bool release_motor(uint8_t motor_id){
...
+  // at a smaller distance then degrees can obviously be smaller
+
+  if (degrees < 1) {
+    bound++;
+  }
+  else if(distance/degrees < 25) {
+    bound--;
+  }
+
+  if(bound > 40) {
```
Comment (technight) [inline code/nxt/src/movement.c]: Consider using defines for magic numbers

File: code/nxt/src/movement.c [hunk 6]
```diff
@@ -128,16 +94,72 @@ bool release_motor(uint8_t motor_id){
...
+    bound = 30;
+  }
+  if(bound > 100) { // overflow
+    bound = 0;
+  }
+  return bound;
+}
+
+int readjust_lower_bound(T_TARGET_LOCATION target){
```
Comment (technight) [inline code/nxt/src/movement.c]: set to return int, but no return in function

File: code/nxt/inc/movement.h [hunk 7]
```diff
@@ -2,22 +2,41 @@
...
+
+#define MOTOR_DEADZONE 10
+
 #define MAX_INPUT_VALUE 127
 
+
+extern uint8_t x_lower_bound_modifier;
+extern uint8_t y_lower_bound_modifier;
+
 // actual methods
```
Comment (Moesgaarda) [inline code/nxt/inc/movement.h]: fjern den her pls

File: code/nxt/inc/movement.h [hunk 8]
```diff
@@ -5,19 +5,40 @@
...
 extern void stop_motors();
-void move_motor(uint8_t motor_id, int32_t speed);
-int get_speed_by_distance(int distance);
+extern int8_t get_speed_by_distance(int8_t distance, char axis);
+extern T_TARGET_LOCATION get_current_location();
+
+extern void readjust_lower_bound(T_TARGET_LOCATION target);
+
+
+bool is_moving_too_slow(int degrees, int distance);
```
Comment (technight) [inline code/nxt/inc/movement.h]: should not be int, same below

**Pull request merged**